### PR TITLE
add auth to get template from repo for private repositories

### PR DESCRIPTION
**Pull request by dim-awsdev1**
## What did you implement:

I added ability to create service from template served in private repository using basic authorization.
for example:

serverless create --template-url https://${user}:${password}@github.com/${owner}/${repo}/${folder}  --path myService

## How did you implement it:

I just added auth param to reterned object of parseGitHubURL and parseBitBucketURL methods in lib/utils/downloadTemplateFromRepo.js file. 
And then include it into "options" parameter of "download" function call

## How can we verify it:

before changes:
master

after changes:
my

## Todos:

- [ ] Write tests
- [ ] Write documentation
- [ ] Fix linting errors
- [ ] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

**Comment by HyperBrain** [comment]
Hey @dim-awsdev1 ,

thanks for the PR. But could you fill out the description template. It is very important for us to know exactly, what the problem is, that you solved - why and how you implemented it and last, how we can verify that the implementation works (so that we know what else to check for side-effects).
Thank you.

**Comment by horike37** [comment]
Hi @dim-awsdev1, any updates?
Let us know if you need any help!

**Comment by horike37** [comment]
@dim-awsdev1 
Closing once for now since it took a long time without any activities.
Ping us if you want to continue working on this.

**Comment by dim-awsdev1** [comment]
Hi @horike37 
I don't  understand why you closed my PR,
@HyperBrain asked me to fill out the description template
I did it, and he change PR stage to in-review from needs-feedback
What  should i do anything else to merge my PR?

**Comment by horike37** [comment]
Thank you for pinging @dim-awsdev1 👍 

>@HyperBrain asked me to fill out the description template
>I did it, and he change PR stage to in-review from needs-feedback
Sorry, I missed that 😅 
just re-opened and will review again!

File: lib/utils/downloadTemplateFromRepo.test.js [hunk 1]
```diff
@@ -146,9 +146,10 @@ describe('downloadTemplateFromRepo', () => {
     });
 
     it('should parse a valid GitHub URL', () => {
-      const output = parseRepoURL('https://github.com/serverless/serverless');
+      const output = parseRepoURL('https://username@github.com/serverless/serverless');
```
Comment (horike37) [inline lib/utils/downloadTemplateFromRepo.test.js]: You added a test case to the existing one, but could you separate private repo case from this one.
We should explicitly support both of private and public.

File: lib/utils/downloadTemplateFromRepo.test.js [hunk 2]
```diff
@@ -159,9 +160,10 @@ describe('downloadTemplateFromRepo', () => {
     });
 
     it('should parse a valid GitHub URL with subdirectory', () => {
-      const output = parseRepoURL('https://github.com/serverless/serverless/tree/master/assets');
+      const output = parseRepoURL('https://username@github.com/serverless/serverless/tree/master/assets');
```
Comment (horike37) [inline lib/utils/downloadTemplateFromRepo.test.js]: Same here.

File: lib/utils/downloadTemplateFromRepo.test.js [hunk 3]
```diff
@@ -172,9 +174,10 @@ describe('downloadTemplateFromRepo', () => {
     });
 
     it('should parse a valid BitBucket URL ', () => {
-      const output = parseRepoURL('https://bitbucket.org/atlassian/localstack');
+      const output = parseRepoURL('https://username@bitbucket.org/atlassian/localstack');
```
Comment (horike37) [inline lib/utils/downloadTemplateFromRepo.test.js]: Same here.

File: lib/utils/downloadTemplateFromRepo.test.js [hunk 4]
```diff
@@ -185,9 +188,10 @@ describe('downloadTemplateFromRepo', () => {
     });
 
     it('should parse a valid BitBucket URL with subdirectory', () => {
-      const output = parseRepoURL('https://bitbucket.org/atlassian/localstack/src/85870856fd6941ae75c0fa946a51cf756ff2f53a/localstack/dashboard/?at=mvn');
+      const output = parseRepoURL('https://username@bitbucket.org/atlassian/localstack/src/85870856fd6941ae75c0fa946a51cf756ff2f53a/localstack/dashboard/?at=mvn');
```
Comment (horike37) [inline lib/utils/downloadTemplateFromRepo.test.js]: Same here.

**Comment by dim-awsdev1** [comment]
Hi @horike37,
Thank you for review.
Unfortunately i don't have private repo which  i can provide to testing. Also it can brings to problem with tests if this repo user just change password or token. 

About documentation: should i change this file [service.md](https://github.com/serverless/serverless/blob/2f658ed9721501747da3f3ef8e5f87e9e15add44/docs/providers/aws/guide/services.md)? 
it have creation section and [create command docs](https://github.com/serverless/serverless/blob/2f658ed9721501747da3f3ef8e5f87e9e15add44/docs/providers/aws/cli-reference/create) link. But i got 404 error when have tried to follow it

**Comment by horike37** [comment]
@dim-awsdev1 
Sorry for the delay response.

>About documentation: should i change this file service.md?
>it have creation section and create command docs link. But i got 404 error when have tried to follow it

Would be good to add this functionality to the `install` command doc. You can also install `serverless install --url https://github.com/some/service` instead of `sls create` command and the `install` command doc has been already written about GitHub examples

https://github.com/serverless/serverless/blob/master/docs/providers/aws/cli-reference/install.md
https://github.com/serverless/serverless/blob/master/docs/providers/azure/cli-reference/install.md
https://github.com/serverless/serverless/blob/master/docs/providers/azure/cli-reference/install.md
https://github.com/serverless/serverless/blob/master/docs/providers/azure/cli-reference/install.md
https://github.com/serverless/serverless/blob/master/docs/providers/azure/cli-reference/install.md
google

**Pull request closed without merging**